src/http/method.zig

```diff
@@ -52,18 +52,19 @@ pub const Method = enum {
 
     const with_body: std.enums.EnumSet(Method) = brk: {
         var values = std.enums.EnumSet(Method).initFull();
+        values.remove(.CONNECT);
         values.remove(.HEAD);
         values.remove(.TRACE);
-        values.remove(.OPTIONS);
         break :brk values;
     };
 
     const with_request_body: std.enums.EnumSet(Method) = brk: {
         var values = std.enums.EnumSet(Method).initFull();
+        values.remove(.CONNECT);
+        values.remove(.GET);
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/GET says 
<img width="776" alt="Screenshot 2024-11-13 at 12 45 34" src="https://github.com/user-attachments/assets/26821101-5d18-4eff-be6a-333454d2fe4c">

Reply:
Whats your point here? GET is removed to have a request body - so thats already correct in my PR? or are you reffering to 
>  A payload within a GET request message has no defined semantics;
   sending a payload body on a GET request might cause some existing
   implementations to reject the request.

how shall we handle this then?

Reply:
for `node:http` and `node:https` we should have the same behavior that node.js has, for Bun.serve and fetch we should respect the https://fetch.spec.whatwg.org/ (you can test the behavior with chrome for example) we will need tests that pass in both (node and bun) to avoid regressions.

Reply:
Chrome: 
> Promise {<rejected>: TypeError: Failed to execute 'fetch' on 'Window': Request with GET/HEAD method cannot have body.
  …}

Node:
> Uncaught TypeError: Request with GET/HEAD method cannot have body.

Reply:
node:http // node:https is ignoring the body in the request and not sending it without warnings/errors